src/clj/puppetlabs/services/config/puppet_server_config_core.clj

```diff
@@ -92,6 +93,18 @@
                     "conflict with the values to be "
                     "read from puppet: " key-conflicts))))))
 
+(defn check-master-url-prefix
+  "Ensure that Puppet Server's route for the puppet master matches what is in
+  puppet.conf."
+  [webrouter-config puppet-config]
+  (let [puppet-prefix (:master-url-prefix puppet-config)
+        webrouter-prefix (:puppetlabs.services.master.master-service/master-service webrouter-config)]
```

Comment:
I think you could use this:

https://github.com/puppetlabs/trapperkeeper/blob/master/src/puppetlabs/trapperkeeper/services.clj#L25

if you did it directly in the `init` definition.  It looks like we don't currently have a protocol for the master service, so you'd have to create one (can be empty), and call it `PuppetMasterService` or whatever.  Then, in the `init` for PuppetServerConfigService you could do something like this:

`(get-service this :PuppetMasterService)`

and pass that to the webrouter `get-route` function.

That ^^ is pretty gross too, but probably a little safer than what you have here, because theoretically the webrouter config could be nested a little more deeply than how we currently have it configured, and if that happened then the approach of trying to pluck the value directly out of the config would either not work or would require a bunch of extra conditional logic.